src/server.c

```diff
@@ -910,10 +910,18 @@ struct redisCommand redisCommandTable[] = {
      "no-script may-replicate @scripting",
      0,evalGetKeys,0,0,0,0,0,0},
 
+    {"eval_ro",evalCommand,-3,
+     "no-script read-only @scripting",
```

Comment:
Having this be marked read-only makes it interact a little bit weirdly with client tracking, since it might track extra keys that wouldn't be tracked for a non readonly script. I think this should maybe just be "no-script @scripting". Maybe worth writing a test to validate it.